packages/mui-joy/src/Button/Button.tsx

```diff
@@ -1,21 +1,250 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_capitalize as capitalize, unstable_useForkRef as useForkRef } from '@mui/utils';
+import { useButton } from '@mui/base/ButtonUnstyled';
```

Comment:
Interesting! So we use the hook, not the unstyled component. This might speak a lot.

Reply:
There is one constraint. If I use ButtonUnstyled, how can the consumer override `component` prop like

```js
<Button component={Link} />
```

another thing is that using ButtonUnstyled will get `ButtonUnstyled-root` className which I think is redundant.

Reply:
@michaldudak Do you have any thoughts about this?

Reply:
I think the comment from @michaldudak answers my question. https://github.com/mui-org/material-ui/pull/29597#issuecomment-968804988

I prefer to use hooks in the design systems.